File: tasks/javascript-regression-014/verify.mjs
import { parseBoolean } from "./parseBoolean.js";

for (const value of ["true", "TRUE", "yes", "1"]) {
  if (parseBoolean(value) !== true) throw new Error(`${value} must be true`);
}
for (const value of ["false", "FALSE", "no", "0"]) {
  if (parseBoolean(value) !== false) throw new Error(`${value} must be false`);
}
try {
  parseBoolean("maybe");
  throw new Error("unknown value must throw TypeError");
} catch (error) {
  if (!(error instanceof TypeError)) throw error;
}

File: tasks/python-regression-004/verify.py
"""Verifier for python-regression-004."""
from parsing import parse_bool

for raw in ("true", "TRUE", "yes", "1"):
    if parse_bool(raw) is not True:
        raise SystemExit(f"{raw!r} must parse as true")
for raw in ("false", "FALSE", "no", "0"):
    if parse_bool(raw) is not False:
        raise SystemExit(f"{raw!r} must parse as false")
try:
    parse_bool("maybe")
except ValueError:
    pass
else:
    raise SystemExit("unknown values must raise ValueError")
